File: gift_list_api/src/main/java/br/com/project/gift_list_api/Gift.java
package br.com.project.gift_list_api;

import jakarta.persistence.*;

@Entity
@Table(name = "gift")
public class Gift {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Float id;
    private String gift_name;
    private boolean selected;
    private String gift_giver;
    private String category;

    public float getId() {
        return id;
    }

    public void setId(float id) {
        this.id = id;
    }

    public String getGift_name() {
        return gift_name;
    }

    public void setGift_name(String gift_name) {
        this.gift_name = gift_name;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public String getGift_giver() {
        return gift_giver;
    }

    public void setGift_giver(String gift_giver) {
        this.gift_giver = gift_giver;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }
}

File: gift_list_front/src/entity/gift.ts
export class Gift {
  id: number;
  gift_name: string;
  selected: boolean;
  gift_giver: string;
  category: string;

  constructor(
    id: number,
    gift_name: string,
    selected: boolean,
    gift_giver: string,
    category: string
  ) {
    this.id = id;
    this.gift_name = gift_name;
    this.selected = selected;
    this.gift_giver = gift_giver;
    this.category = category;
  }
}
